src/Workspaces/Core/Portable/Differencing/SequenceEdit.cs

```diff
@@ -18,40 +18,26 @@ internal struct SequenceEdit : IEquatable<SequenceEdit>
     {
         private readonly int _oldIndex;
         private readonly int _newIndex;
+        private readonly EditKind _kind;
 
-        internal SequenceEdit(int oldIndex, int newIndex)
+        internal SequenceEdit(int oldIndex, int newIndex, EditKind kind)
         {
             Debug.Assert(oldIndex >= -1);
             Debug.Assert(newIndex >= -1);
             Debug.Assert(newIndex != -1 || oldIndex != -1);
 
             _oldIndex = oldIndex;
             _newIndex = newIndex;
+            _kind = kind;
         }
 
         /// <summary>
         /// The kind of edit: <see cref="EditKind.Delete"/>, <see cref="EditKind.Insert"/>, or <see cref="EditKind.Update"/>.
         /// </summary>
-        public EditKind Kind
-        {
-            get
-            {
-                if (_oldIndex == -1)
-                {
-                    return EditKind.Insert;
-                }
-
-                if (_newIndex == -1)
-                {
-                    return EditKind.Delete;
-                }
-
-                return EditKind.Update;
-            }
-        }
+        public EditKind Kind => _kind;
 
         /// <summary>
-        /// Index in the old sequence, or -1 if the edit is insert.
+        /// Index in the old sequence.
         /// </summary>
         public int OldIndex => _oldIndex;
```

Comment:
>OldIndex [](http://example.com/codeflow?start=19&length=8)

What is the exact semantics of OldIndex value for Insert?

Say I have sequences: (a, b, b, c) and (a, c, x). 
Is the OldIndex of insert of x expected to be 4?

Reply:
Setting OldIndex on Insert edit seems unnecessary as it can be easily inferred from the Update operation that precedes it. 

For the above example the edits will be:

U[0,0], D[1], D[2], U[3,1], D[4], I[2], U[5,3].

The last Update before I[2] is U[3,1]. Depending on whether you want to count deletes or not the desired "old index" for I is either 5 or 4 (4 + 1, 3 + 1)